File: 217-contains-duplicate/contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        // Bruteforce: O(nlogn + n)
//         Arrays.sort(nums);
        
//         for(int i = 1; i < nums.length; i ++){
//             if(nums[i] == nums[i - 1])
//                 return true;
//         }
        
//         return false;
        HashMap<Integer, Integer> map = new HashMap<>();
        
        for(Integer i: nums)
            map.put(i, map.getOrDefault(i, 0) + 1);
        
        for(Map.Entry<Integer,Integer> entry: map.entrySet()){
            if(entry.getValue() > 1)
                return true;
        }
        
        return false;
    }
}

File: 217-contains-duplicate/contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        if len(nums) == 1:
            return False
        without_duplicates = set()
        for num in nums:
            # if num in without_duplicates:
            #     return True
            without_duplicates.add(num)
        return False if len(nums) == len(without_duplicates) else True